In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [2]:
sale_info = pd.read_csv('../../data/raw/EXTR_RPSale.csv')
sale_info.head(2)

ExciseTaxNbr   Major  Minor DocumentDate  SalePrice    RecordingNbr Volume  \
0       2857854  198920   1430   03/28/2017          0  20170410000541          
1       2743355  638580    110   07/14/2015     190000  20150715002686          

  Page PlatNbr PlatType  ... PropertyType PrincipalUse SaleInstrument  \
0                        ...            3            7             15   
1                        ...            3            6              3   

  AFForestLand  AFCurrentUseLand  AFNonProfitUse  AFHistoricProperty  \
0            N                 N               N                   N   
1            N                 N               N                   N   

  SaleReason PropertyClass SaleWarning  
0         16             2       20 31  
1          1             8          15  

[2 rows x 24 columns]

In [3]:
sale_info = sale_info[['Major', 'Minor', 'SalePrice', 'DocumentDate','PrincipalUse']]
sale_info.head()

Major  Minor  SalePrice DocumentDate  PrincipalUse
0  198920   1430          0   03/28/2017             7
1  638580    110     190000   07/14/2015             6
2  919715    200     192000   07/08/2019             2
3  894677    240     818161   12/21/2016             6
4  445872    260          0   10/03/2016             2

In [4]:
sale_info['PrincipalUse'].value_counts()

6     266390
2      63230
7      18933
11       856
9        442
4        441
0        371
10       350
3         20
1         13
8         12
5          9
Name: PrincipalUse, dtype: int64

In [5]:
#Only using data related to residential properties. According to the LookUp file, '6' represents a represential property.
sale_info = sale_info[sale_info['PrincipalUse'] == 6] #filtering out
sale_info.drop(['PrincipalUse'], axis=1, inplace=True) #removing the column because now that the data contains only
#residential properties, we have no use for it to be labeled.
sale_info.head()

Major  Minor  SalePrice DocumentDate
1  638580    110     190000   07/14/2015
3  894677    240     818161   12/21/2016
5  408330   4150          0   03/15/2017
6  510140   4256          0   07/25/2016
7  126320    110     375000   10/30/2017

In [6]:
len(sale_info)

266390

In [7]:
sale_info['SalePrice'].describe()

count    2.663900e+05
mean     5.273181e+05
std      1.099445e+06
min     -6.000000e+02
25%      0.000000e+00
50%      3.999500e+05
75%      7.000000e+05
max      6.250000e+07
Name: SalePrice, dtype: float64

In [8]:
sale_info['SalePrice'].value_counts()

0          83463
450000      1372
550000      1297
400000      1278
650000      1232
           ...  
320001         1
381479         1
1008241        1
565897         1
2430           1
Name: SalePrice, Length: 22774, dtype: int64

Given that there are 266390 entries in this data frame and 25% of the 'SalePrice' column is 0 (83463 entries), I've decided to drop the rows with no sale price, as replacing missing values with the column mean could greatly skew the data and adversely effect the results.

In [9]:
sale_info.drop(sale_info.loc[sale_info['SalePrice']==0].index, inplace=True)
sale_info['SalePrice'].value_counts()

450000     1372
550000     1297
400000     1278
650000     1232
600000     1214
           ... 
464815        1
1560500       1
237502        1
868330        1
1851400       1
Name: SalePrice, Length: 22773, dtype: int64

In [10]:
len(sale_info)

182927

In [11]:
sale_info.head()

Major  Minor  SalePrice DocumentDate
1   638580    110     190000   07/14/2015
3   894677    240     818161   12/21/2016
7   126320    110     375000   10/30/2017
9   213043    120     560000   12/20/2019
10  940652    630     435000   07/22/2019

In [36]:
housing_data = pd.read_csv('../../data/raw/EXTR_ResBldg.csv')
housing_data.head(2)

Major  Minor  BldgNbr  NbrLivingUnits                     Address  \
0   9800    720        1               1  27719  SE 26TH WAY   98075   
1   9802    140        1               1  2829   277TH TER SE  98075   

  BuildingNumber Fraction DirectionPrefix                 StreetName  \
0          27719                       SE  26TH                        
1          2829                            277TH                       

  StreetType  ... FpMultiStory FpFreestanding  FpAdditional  YrBuilt  \
0     WAY     ...            0              0             0     2001   
1     TER     ...            0              0             0     2004   

   YrRenovated  PcntComplete  Obsolescence  PcntNetCondition  Condition  \
0            0             0             0                 0          3   
1            0             0             0                 0          3   

   AddnlCost  
0          0  
1          0  

[2 rows x 50 columns]

In [40]:
housing_data['FinBasementGrade'].value_counts()

0     108010
7      28497
8      16246
6      16218
9       6283
5       2983
10      1834
11       725
4        436
12       150
3         73
13        37
2         14
1          3
20         1
Name: FinBasementGrade, dtype: int64

In [13]:
to_drop2 = ['DirectionPrefix', 'DirectionSuffix', 'ZipCode', 'Fraction', 'FpFreestanding', 'FpAdditional',
                  'YrRenovated', 'PcntComplete', 'PcntNetCondition', 'AddnlCost','FpMultiStory', 'Obsolescence',
                  'ViewUtilization', 'FpSingleStory','BrickStone', 'FpSingleStory', 'FinBasementGrade',
                   'StreetName', 'StreetType', 'BldgGradeVar', 'SqFtHalfFloor', 'SqFtUnfinFull', 'SqFtUnfinHalf', 'BldgNbr',
                   'NbrLivingUnits', 'Address', 'BuildingNumber', 'SqFtDeck', 'DaylightBasement']
housing_data.drop(to_drop2, axis= 1, inplace=True)

In [14]:
housing_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181510 entries, 0 to 181509
Data columns (total 22 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Major               181510 non-null  int64  
 1   Minor               181510 non-null  int64  
 2   Stories             181510 non-null  float64
 3   BldgGrade           181510 non-null  int64  
 4   SqFt1stFloor        181510 non-null  int64  
 5   SqFt2ndFloor        181510 non-null  int64  
 6   SqFtUpperFloor      181510 non-null  int64  
 7   SqFtTotLiving       181510 non-null  int64  
 8   SqFtTotBasement     181510 non-null  int64  
 9   SqFtFinBasement     181510 non-null  int64  
 10  SqFtGarageBasement  181510 non-null  int64  
 11  SqFtGarageAttached  181510 non-null  int64  
 12  SqFtOpenPorch       181510 non-null  int64  
 13  SqFtEnclosedPorch   181510 non-null  int64  
 14  HeatSystem          181510 non-null  int64  
 15  HeatSource          181510 non-nul

In [15]:
#For the sake of streamlining, I will merge the bath columns and combine number of bathrooms and create a column called 
#'Bathrooms' with the total amount of bathrooms (regardless of its size), per home.
housing_data['Bathrooms'] = housing_data[['BathFullCount', 'BathHalfCount', 'Bath3qtrCount']].astype(float).sum(1)
to_drop= ['BathFullCount', 'BathHalfCount', 'Bath3qtrCount']
housing_data.drop(to_drop, axis=1, inplace=True)
housing_data.head()

Major  Minor  Stories  BldgGrade  SqFt1stFloor  SqFt2ndFloor  \
0   9800    720      2.0         11          1970          2130   
1   9802    140      2.0         10          1610          1400   
2   9830     20      2.0         10          2520          2560   
3   9830    160      2.0         10          2210          1860   
4  10050    180      2.0          7           910           700   

   SqFtUpperFloor  SqFtTotLiving  SqFtTotBasement  SqFtFinBasement  \
0               0           4100                0                0   
1               0           3010                0                0   
2               0           5080                0                0   
3               0           4070                0                0   
4               0           1610                0                0   

   SqFtGarageBasement  SqFtGarageAttached  SqFtOpenPorch  SqFtEnclosedPorch  \
0                   0                 750              0                  0   
1                   0                 660            380                  0   
2                   0                1020            360                  0   
3                   0                1000            690                  0   
4                   0                 440             60                  0   

   HeatSystem  HeatSource  Bedrooms  YrBuilt  Condition  Bathrooms  
0           5           2         4     2001          3        4.0  
1           5           2         4     2004          3        5.0  
2           5           2         4     2017          3        5.0  
3           5           2         4     2013          3        4.0  
4           5           2         3     1994          4        3.0

In [16]:
housing_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181510 entries, 0 to 181509
Data columns (total 20 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Major               181510 non-null  int64  
 1   Minor               181510 non-null  int64  
 2   Stories             181510 non-null  float64
 3   BldgGrade           181510 non-null  int64  
 4   SqFt1stFloor        181510 non-null  int64  
 5   SqFt2ndFloor        181510 non-null  int64  
 6   SqFtUpperFloor      181510 non-null  int64  
 7   SqFtTotLiving       181510 non-null  int64  
 8   SqFtTotBasement     181510 non-null  int64  
 9   SqFtFinBasement     181510 non-null  int64  
 10  SqFtGarageBasement  181510 non-null  int64  
 11  SqFtGarageAttached  181510 non-null  int64  
 12  SqFtOpenPorch       181510 non-null  int64  
 13  SqFtEnclosedPorch   181510 non-null  int64  
 14  HeatSystem          181510 non-null  int64  
 15  HeatSource          181510 non-nul

In [17]:
#Moving 'Bathrooms' column to just after 'Bathrooms'
bath = housing_data['Bathrooms']
housing_data.drop(labels=['Bathrooms'], axis=1, inplace = True)
housing_data.insert(17, 'Bathrooms', bath) #<-- '17' represents the column index number that I want to assign to this column.
#The column just after 'Bedrooms' is 'YrBuilt' whose index number is 17.
housing_data.head()

Major  Minor  Stories  BldgGrade  SqFt1stFloor  SqFt2ndFloor  \
0   9800    720      2.0         11          1970          2130   
1   9802    140      2.0         10          1610          1400   
2   9830     20      2.0         10          2520          2560   
3   9830    160      2.0         10          2210          1860   
4  10050    180      2.0          7           910           700   

   SqFtUpperFloor  SqFtTotLiving  SqFtTotBasement  SqFtFinBasement  \
0               0           4100                0                0   
1               0           3010                0                0   
2               0           5080                0                0   
3               0           4070                0                0   
4               0           1610                0                0   

   SqFtGarageBasement  SqFtGarageAttached  SqFtOpenPorch  SqFtEnclosedPorch  \
0                   0                 750              0                  0   
1                   0                 660            380                  0   
2                   0                1020            360                  0   
3                   0                1000            690                  0   
4                   0                 440             60                  0   

   HeatSystem  HeatSource  Bedrooms  Bathrooms  YrBuilt  Condition  
0           5           2         4        4.0     2001          3  
1           5           2         4        5.0     2004          3  
2           5           2         4        5.0     2017          3  
3           5           2         4        4.0     2013          3  
4           5           2         3        3.0     1994          4

In [18]:
parcel = pd.read_csv('../../data/raw/EXTR_Parcel.csv')
parcel.head(2)

Unnamed: 0   Major  Minor PropName                            PlatName  \
0           0  807841    410                      SUMMER RIDGE DIV NO. 02   
1           2  755080     15           SANDER'S TO GILMAN PK & SALMON BAY   

          PlatLot PlatBlock  Range  Township  Section  ... SeismicHazard  \
0  41                            6        25       22  ...             N   
1  3                1            3        25       11  ...             N   

  LandslideHazard  SteepSlopeHazard  Stream  Wetland  SpeciesOfConcern  \
0               N                 N       N        N                 N   
1               N                 N       N        N                 N   

  SensitiveAreaTract  WaterProblems TranspConcurrency  OtherProblems  
0                  N              N                 N              N  
1                  N              N                 N              N  

[2 rows x 82 columns]

In [19]:
parcel = parcel[['Major', 'Minor', 'SqFtLot', 'TrafficNoise','OtherNuisances']]
parcel.head()

Major  Minor  SqFtLot  TrafficNoise OtherNuisances
0  807841    410     7424             0              N
1  755080     15     5000             0              N
2  888600    135   277041             0              N
3   22603   9181    10560             0              N
4  229670    160     9853             0              N

In [20]:
display(parcel['TrafficNoise'].value_counts())
display("Traffic noise ranges on a scale of 0 to 3, 0 is no noise, 1 being moderate, 2 being high, and 3 being extreme")

0    176076
1     16736
2     10563
3      1824
Name: TrafficNoise, dtype: int64

'Traffic noise ranges on a scale of 0 to 3, 0 is no noise, 1 being moderate, 2 being high, and 3 being extreme'

In [21]:
parcel['OtherNuisances'].value_counts()

N    199057
Y      6142
Name: OtherNuisances, dtype: int64

In [22]:
#Replacing 'N' with 0 and 'Y' with 1 (No/Yes) for easy plotting
parcel['OtherNuisances']=parcel['OtherNuisances'].replace('N',0)
parcel['OtherNuisances']=parcel['OtherNuisances'].replace('Y',1)
parcel['OtherNuisances'].value_counts()

0    199057
1      6142
Name: OtherNuisances, dtype: int64

I'll now merge my data frames for one clean file:

In [23]:
house_data = pd.merge(sale_info, housing_data,  how='left', left_on=['Major','Minor'], right_on = ['Major','Minor'])
house_data.head()

Major  Minor  SalePrice DocumentDate  Stories  BldgGrade  SqFt1stFloor  \
0  638580    110     190000   07/14/2015      1.0        7.0        1080.0   
1  894677    240     818161   12/21/2016      2.0        9.0        1020.0   
2  126320    110     375000   10/30/2017      2.0        6.0        1250.0   
3  213043    120     560000   12/20/2019      1.0        7.0        1930.0   
4  940652    630     435000   07/22/2019      2.0        7.0        1057.0   

   SqFt2ndFloor  SqFtUpperFloor  SqFtTotLiving  ...  SqFtGarageBasement  \
0           0.0             0.0         2130.0  ...                 0.0   
1        1430.0             0.0         2450.0  ...                 0.0   
2         400.0             0.0         1650.0  ...                 0.0   
3           0.0             0.0         1930.0  ...                 0.0   
4         918.0             0.0         1975.0  ...                 0.0   

   SqFtGarageAttached  SqFtOpenPorch  SqFtEnclosedPorch  HeatSystem  \
0                 0.0            0.0                0.0         5.0   
1               400.0           40.0                0.0         5.0   
2                 0.0            0.0                0.0         4.0   
3               550.0          180.0                0.0         5.0   
4               465.0           45.0                0.0         5.0   

   HeatSource  Bedrooms  Bathrooms  YrBuilt  Condition  
0         2.0       4.0        2.0   1963.0        3.0  
1         2.0       4.0        3.0   2016.0        3.0  
2         3.0       3.0        1.0   1919.0        3.0  
3         2.0       3.0        2.0   1989.0        3.0  
4         2.0       3.0        3.0   1994.0        3.0  

[5 rows x 22 columns]

In [24]:
house_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 185098 entries, 0 to 185097
Data columns (total 22 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Major               185098 non-null  int64  
 1   Minor               185098 non-null  int64  
 2   SalePrice           185098 non-null  int64  
 3   DocumentDate        185098 non-null  object 
 4   Stories             173142 non-null  float64
 5   BldgGrade           173142 non-null  float64
 6   SqFt1stFloor        173142 non-null  float64
 7   SqFt2ndFloor        173142 non-null  float64
 8   SqFtUpperFloor      173142 non-null  float64
 9   SqFtTotLiving       173142 non-null  float64
 10  SqFtTotBasement     173142 non-null  float64
 11  SqFtFinBasement     173142 non-null  float64
 12  SqFtGarageBasement  173142 non-null  float64
 13  SqFtGarageAttached  173142 non-null  float64
 14  SqFtOpenPorch       173142 non-null  float64
 15  SqFtEnclosedPorch   173142 non-nul

In [25]:
house_data = pd.merge(house_data, parcel,  how='left', left_on=['Major','Minor'], right_on = ['Major','Minor'])
house_data.head()

Major  Minor  SalePrice DocumentDate  Stories  BldgGrade  SqFt1stFloor  \
0  638580    110     190000   07/14/2015      1.0        7.0        1080.0   
1  894677    240     818161   12/21/2016      2.0        9.0        1020.0   
2  126320    110     375000   10/30/2017      2.0        6.0        1250.0   
3  213043    120     560000   12/20/2019      1.0        7.0        1930.0   
4  940652    630     435000   07/22/2019      2.0        7.0        1057.0   

   SqFt2ndFloor  SqFtUpperFloor  SqFtTotLiving  ...  SqFtEnclosedPorch  \
0           0.0             0.0         2130.0  ...                0.0   
1        1430.0             0.0         2450.0  ...                0.0   
2         400.0             0.0         1650.0  ...                0.0   
3           0.0             0.0         1930.0  ...                0.0   
4         918.0             0.0         1975.0  ...                0.0   

   HeatSystem  HeatSource  Bedrooms  Bathrooms  YrBuilt  Condition  SqFtLot  \
0         5.0         2.0       4.0        2.0   1963.0        3.0  10534.0   
1         5.0         2.0       4.0        3.0   2016.0        3.0   3813.0   
2         4.0         3.0       3.0        1.0   1919.0        3.0   4760.0   
3         5.0         2.0       3.0        2.0   1989.0        3.0  10523.0   
4         5.0         2.0       3.0        3.0   1994.0        3.0   7392.0   

   TrafficNoise  OtherNuisances  
0           0.0             0.0  
1           0.0             0.0  
2           0.0             0.0  
3           0.0             0.0  
4           0.0             0.0  

[5 rows x 25 columns]

In [26]:
house_data.describe()

Major          Minor     SalePrice        Stories  \
count  185098.000000  185098.000000  1.850980e+05  173142.000000   
mean   444596.198041    1706.125307  7.705447e+05       1.552041   
std    287964.570067    2995.320781  1.265627e+06       0.541692   
min       100.000000       1.000000 -6.000000e+02       1.000000   
25%    198020.000000     120.000000  3.840000e+05       1.000000   
50%    375160.000000     340.000000  5.750000e+05       1.500000   
75%    722850.000000    1140.000000  8.350000e+05       2.000000   
max    990600.000000    9694.000000  6.250000e+07       4.000000   

           BldgGrade   SqFt1stFloor   SqFt2ndFloor  SqFtUpperFloor  \
count  173142.000000  173142.000000  173142.000000   173142.000000   
mean        7.733375    1265.010356     567.337942       15.839196   
std         1.199448     509.909620     691.403320       99.391761   
min         1.000000       0.000000       0.000000        0.000000   
25%         7.000000     950.000000       0.000000        0.000000   
50%         8.000000    1210.000000       0.000000        0.000000   
75%         8.000000    1500.000000    1120.000000        0.000000   
max        13.000000    8760.000000    6820.000000     2590.000000   

       SqFtTotLiving  SqFtTotBasement  ...  SqFtEnclosedPorch     HeatSystem  \
count  173142.000000    173142.000000  ...      173142.000000  173142.000000   
mean     2194.650899       469.087125  ...           5.857995       4.970117   
std      1009.484566       582.105029  ...          39.414054       1.076078   
min         1.000000         0.000000  ...           0.000000       0.000000   
25%      1470.000000         0.000000  ...           0.000000       5.000000   
50%      2000.000000         0.000000  ...           0.000000       5.000000   
75%      2730.000000       940.000000  ...           0.000000       5.000000   
max     14980.000000      5610.000000  ...        2330.000000       8.000000   

          HeatSource       Bedrooms      Bathrooms        YrBuilt  \
count  173142.000000  173142.000000  173142.000000  173142.000000   
mean        2.112740       3.460535       2.640093    1977.452403   
std         0.571527       0.975727       1.045970      31.538980   
min         0.000000       0.000000       0.000000    1900.000000   
25%         2.000000       3.000000       2.000000    1955.000000   
50%         2.000000       3.000000       3.000000    1980.000000   
75%         2.000000       4.000000       3.000000    2006.000000   
max         7.000000      24.000000      16.000000    2020.000000   

           Condition       SqFtLot   TrafficNoise  OtherNuisances  
count  173142.000000  1.839260e+05  183926.000000   183926.000000  
mean        3.464099  4.346159e+04       0.221828        0.031768  
std         0.709047  7.090436e+05       0.578672        0.175383  
min         1.000000  0.000000e+00       0.000000        0.000000  
25%         3.000000  5.000000e+03       0.000000        0.000000  
50%         3.000000  7.592000e+03       0.000000        0.000000  
75%         4.000000  1.120000e+04       0.000000        0.000000  
max         5.000000  2.330525e+07       3.000000        1.000000  

[8 rows x 24 columns]

In [27]:
#Checking for any missing data:
for column in list(house_data.columns):
    print(column, sum(house_data[column].isnull()))

Major 0
Minor 0
SalePrice 0
DocumentDate 0
Stories 11956
BldgGrade 11956
SqFt1stFloor 11956
SqFt2ndFloor 11956
SqFtUpperFloor 11956
SqFtTotLiving 11956
SqFtTotBasement 11956
SqFtFinBasement 11956
SqFtGarageBasement 11956
SqFtGarageAttached 11956
SqFtOpenPorch 11956
SqFtEnclosedPorch 11956
HeatSystem 11956
HeatSource 11956
Bedrooms 11956
Bathrooms 11956
YrBuilt 11956
Condition 11956
SqFtLot 1172
TrafficNoise 1172
OtherNuisances 1172


A number of the columns have missing data and to attempt to replace the null values with the median is not feasible (the sqft of a lot varies greatly and can't be averaged out), so I will rows containing empty values:

In [28]:
house_data = house_data.dropna(how='any',axis=0)
house_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 173142 entries, 0 to 185097
Data columns (total 25 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Major               173142 non-null  int64  
 1   Minor               173142 non-null  int64  
 2   SalePrice           173142 non-null  int64  
 3   DocumentDate        173142 non-null  object 
 4   Stories             173142 non-null  float64
 5   BldgGrade           173142 non-null  float64
 6   SqFt1stFloor        173142 non-null  float64
 7   SqFt2ndFloor        173142 non-null  float64
 8   SqFtUpperFloor      173142 non-null  float64
 9   SqFtTotLiving       173142 non-null  float64
 10  SqFtTotBasement     173142 non-null  float64
 11  SqFtFinBasement     173142 non-null  float64
 12  SqFtGarageBasement  173142 non-null  float64
 13  SqFtGarageAttached  173142 non-null  float64
 14  SqFtOpenPorch       173142 non-null  float64
 15  SqFtEnclosedPorch   173142 non-nul

In [29]:
#Rechecking for any missing data:
for column in list(house_data.columns):
    print(column, sum(house_data[column].isnull()))

Major 0
Minor 0
SalePrice 0
DocumentDate 0
Stories 0
BldgGrade 0
SqFt1stFloor 0
SqFt2ndFloor 0
SqFtUpperFloor 0
SqFtTotLiving 0
SqFtTotBasement 0
SqFtFinBasement 0
SqFtGarageBasement 0
SqFtGarageAttached 0
SqFtOpenPorch 0
SqFtEnclosedPorch 0
HeatSystem 0
HeatSource 0
Bedrooms 0
Bathrooms 0
YrBuilt 0
Condition 0
SqFtLot 0
TrafficNoise 0
OtherNuisances 0


In [30]:
len(house_data)

173142

In [31]:
house_data.tail(2)

Major  Minor  SalePrice DocumentDate  Stories  BldgGrade  \
185096  769791     30     133000   07/16/2016      2.0        7.0   
185097  678120     47     399950   04/23/2015      2.0        8.0   

        SqFt1stFloor  SqFt2ndFloor  SqFtUpperFloor  SqFtTotLiving  ...  \
185096        1110.0         770.0             0.0         1880.0  ...   
185097         680.0         580.0             0.0         1450.0  ...   

        SqFtEnclosedPorch  HeatSystem  HeatSource  Bedrooms  Bathrooms  \
185096                0.0         5.0         2.0       3.0        3.0   
185097                0.0         5.0         2.0       3.0        2.0   

        YrBuilt  Condition  SqFtLot  TrafficNoise  OtherNuisances  
185096   1990.0        3.0   5936.0           0.0             0.0  
185097   2015.0        3.0   1360.0           0.0             0.0  

[2 rows x 25 columns]

In [32]:
#Resetting the index
house_data.reset_index(drop=True, inplace=True)

In [33]:
house_data.tail(2)

Major  Minor  SalePrice DocumentDate  Stories  BldgGrade  \
173140  769791     30     133000   07/16/2016      2.0        7.0   
173141  678120     47     399950   04/23/2015      2.0        8.0   

        SqFt1stFloor  SqFt2ndFloor  SqFtUpperFloor  SqFtTotLiving  ...  \
173140        1110.0         770.0             0.0         1880.0  ...   
173141         680.0         580.0             0.0         1450.0  ...   

        SqFtEnclosedPorch  HeatSystem  HeatSource  Bedrooms  Bathrooms  \
173140                0.0         5.0         2.0       3.0        3.0   
173141                0.0         5.0         2.0       3.0        2.0   

        YrBuilt  Condition  SqFtLot  TrafficNoise  OtherNuisances  
173140   1990.0        3.0   5936.0           0.0             0.0  
173141   2015.0        3.0   1360.0           0.0             0.0  

[2 rows x 25 columns]

In [41]:
#Saving as a .csv:
house_data = house_data.to_csv('house_data.csv', index = True) 

In [42]:
house_data = pd.read_csv('../../data/processed/house_data.csv')

In [44]:
house_data.tail()

Unnamed: 0   Major  Minor  SalePrice DocumentDate  Stories  BldgGrade  \
173137      173137  438920   1058    1000000   08/20/2015      2.0       10.0   
173138      173138  219331    270     850000   06/07/2018      1.5        8.0   
173139      173139  886030    550     900000   07/16/2016      1.0        8.0   
173140      173140  769791     30     133000   07/16/2016      2.0        7.0   
173141      173141  678120     47     399950   04/23/2015      2.0        8.0   

        SqFt1stFloor  SqFt2ndFloor  SqFtUpperFloor  ...  SqFtEnclosedPorch  \
173137        2160.0        1400.0             0.0  ...                0.0   
173138        1200.0           0.0             0.0  ...                0.0   
173139        1200.0           0.0             0.0  ...                0.0   
173140        1110.0         770.0             0.0  ...                0.0   
173141         680.0         580.0             0.0  ...                0.0   

        HeatSystem  HeatSource  Bedrooms  Bathrooms  YrBuilt  Condition  \
173137         5.0         2.0       5.0        6.0   2016.0        3.0   
173138         5.0         2.0       4.0        3.0   1984.0        3.0   
173139         5.0         2.0       4.0        3.0   1975.0        3.0   
173140         5.0         2.0       3.0        3.0   1990.0        3.0   
173141         5.0         2.0       3.0        2.0   2015.0        3.0   

        SqFtLot  TrafficNoise  OtherNuisances  
173137   7302.0           2.0             0.0  
173138   7832.0           0.0             0.0  
173139  13238.0           0.0             0.0  
173140   5936.0           0.0             0.0  
173141   1360.0           0.0             0.0  

[5 rows x 26 columns]